### ISSUE TITLE
Job steps sometimes execute out of sequence

### ISSUE THREAD

**Issue by ajxb** [rundeck/rundeck#3658]
**Describe the bug**

During execution of a job that runs against multiple nodes in parallel we noticed that the job steps were executed out of order on 1 out of 17 nodes, this caused the job to fail on the given node.

The nodes in question are all Windows 2012 server running cygwin sshd.

**My Rundeck detail**

* Rundeck version: 2.9.3
* install type: rpm
* OS Name/version: CentOS 6.9 x86_64
* DB Type/version: mysql 5.5.45

**To Reproduce**

Unknown if it is possible to easily reproduce, I've only spotted this problem once.

**Expected behavior**

Jobs steps should be executed according to the order in which they are defined in the configuration file consistently across all defined nodes.

**Screenshots**

See highlighted times for node01 indicating that _Copy updated domain payload_ started after _Distribute block files_.  _Distribute block Files_ requires the file copied in the previous step in order to execute successfully.

![results](https://user-images.githubusercontent.com/1756536/42455259-a8a9b79a-8389-11e8-9651-3b52977f78af.png)

**Additional context**

Please see copies of the job definition in yaml, and the log files associated with the execution of this job:

1. [802dd6f5-1e60-4ce1-8a79-aacf41fd3f8e.yaml.txt](https://github.com/rundeck/rundeck/files/2176487/802dd6f5-1e60-4ce1-8a79-aacf41fd3f8e.yaml.txt)
1. [233290.rdlog.txt](https://github.com/rundeck/rundeck/files/2176488/233290.rdlog.txt)
1. [233290.state.json.txt](https://github.com/rundeck/rundeck/files/2176489/233290.state.json.txt)

**Comment by gschueler** [comment]
i'm not certain, but it sounds like it might be caused by bug #2766 and #2756 which is fixed in 2.9.4

**Comment by ajxb** [comment]
Ach that's not good news, on every attempt to upgrade we encounter https://github.com/rundeck/rundeck/issues/2877 and I can't risk the disruption to our production environment as it stands.

**Comment by spaced0gg** [comment]
I have encountered this issue at the weekend.

A job was run against 2 nodes.  On one of the nodes the step to extract the software started out of order. The steps after extracting software did not behave correctly as the software had not extracted when they were started.

![image](https://user-images.githubusercontent.com/16479298/50082341-9ed04300-01e8-11e9-8229-41f1b675d2a6.png)

Rundeck version is 2.10.7-1

**Comment by mrtnmrk** [comment]
Are you sure this issue is solved in 2.9.4? Because we are running 2.11.5 and we are encountering it as well:
![image](https://user-images.githubusercontent.com/5289329/50643409-50ec6680-0f6e-11e9-9a9e-269e78a0ae04.png)

**Comment by puremourning** [comment]
yes it is still a problem

**Comment by ajxb** [comment]
We've recently upgraded to 3.0.13 and are still seeing this exact same problem still.

**Comment by martingoldammer** [comment]
Hi, some progress with this issue?

**Comment by spaced0gg** [comment]
Hi, we have had another occurrence of this issue.  Has there been any progress on resolving this issue?

**Comment by fdevans** [comment]
We apologize if this issue was missed.  If you are still experiencing this issue on the current version of Rundeck please let us know by opening a new Issue request with version and steps to reproduce.  If you are an Enterprise customer our support personnel are ready to help.
Thank you,
The Rundeck Team
